c7n/resolver.py

```diff
@@ -166,20 +167,36 @@ def _get_values(self):
                 res = jmespath.search(self.data['expr'], data)
                 if res is None:
                     log.warning('ValueFrom filter: %s key returned None' % self.data['expr'])
+                if isinstance(res, list):
+                    res = set(res)
                 return res
         elif format == 'csv' or format == 'csv2dict':
             data = csv.reader(io.StringIO(contents))
             if format == 'csv2dict':
                 data = {x[0]: list(x[1:]) for x in zip(*data)}
+                if 'expr' in self.data:
+                    res = jmespath.search(self.data['expr'], data)
```

Comment:
these duplicate blocks look like they could be made into a private helper method